projects/player/app/VoicebasePlayer.js

```
@@ -4,32 +4,35 @@ import { Provider } from 'react-redux'
 import { rootReducers } from './redux/root';
 import DevTools from './redux/utils/DevTools'
 import configureStore from './redux/configureStore';
-import { setBaseUrl } from './api/baseUrl'
 import VoicebasePlayerView from './views/VoicebasePlayerView'
 
 export class VoicebasePlayer extends React.Component {
   static propTypes = {
-    apiUrl: PropTypes.string.isRequired,
-    token: PropTypes.string.isRequired,
-    mediaId: PropTypes.string.isRequired,
+    apiUrl: PropTypes.string,
+    token: PropTypes.string,
+    mediaId: PropTypes.string,
+    analytics: PropTypes.object,
+    mediaUrl: PropTypes.string,
     searchString: PropTypes.string,
     initialStore: PropTypes.object
   };
 
   constructor(props) {
     super(props);
     this.store = configureStore(props.initialStore || {}, rootReducers);
-    setBaseUrl(props.apiUrl);
   }
 
   render () {
-    const { token, mediaId, searchString } = this.props;
+    const { apiUrl, token, mediaId, analytics, mediaUrl, searchString } = this.props;
     return (
       <Provider store={this.store}>
         <div>
           <VoicebasePlayerView
```

Yeah, I think that would be good too as I've already had issues with the player swallowing errors and having to dig pretty hard to find them.

How does this sound:
- The core player and all decorators have an `onError` proptype.func which is optional.
- Decorators can call the onError function if they are unable to get the data required or they find issues while constructing a store if they are required to construct one or any other reason really.
- The core player takes the onError prop and to save passing it down through every component we create a service that is passed by context so any component that may error can call on this context.